core/config/config_helper.cpp

```diff
@@ -113,6 +113,18 @@ parse_minimal_criteria(const pnode& config, const registry& context,
 
     type_descriptor updated_td = update_type(config, td);
 
+    // Although it can still be caught in the following map, it gives consistent
+    // error exception when some keys are not allowed.
+    std::set<std::string> allowed_keys{"time",
```

Comment:
There is potential for typos here - can this be initialized from the map, or collect the string literals in a single place, either a constexpr variable or maybe a macro?